src/test/java/com/mongodb/kafka/connect/sink/StartedMongoSinkTaskTest.java

```diff
@@ -0,0 +1,389 @@
+    client.configureCapturing(DEFAULT_NAMESPACE);
+    StartedMongoSinkTask task =
+        new StartedMongoSinkTask(config, client.mongoClient(), errorReporter);
+    RecordsAndExpectations recordsAndExpectations =
+        new RecordsAndExpectations(
+            asList(
+                Records.simpleValid(TEST_TOPIC, 0),
+                Records.simpleInvalid(TEST_TOPIC, 1),
+                Records.simpleValid(TEST_TOPIC, 2),
+                Records.simpleValid(TEST_TOPIC2, 3)),
+            asList(0, 2, 3),
+            singletonList(1));
+    task.put(recordsAndExpectations.records());
+    recordsAndExpectations.assertExpectations(
+        client.capturedBulkWrites().get(DEFAULT_NAMESPACE),
+        errorReporter.reported(),
+        Records::match);
+  }
+
+  @Test
+  void putTolerateNonePostProcessingError() {
+    properties.put(MongoSinkTopicConfig.ERRORS_TOLERANCE_CONFIG, ErrorTolerance.NONE.value());
+    MongoSinkConfig config = new MongoSinkConfig(properties);
+    client.configureCapturing(DEFAULT_NAMESPACE);
+    StartedMongoSinkTask task =
+        new StartedMongoSinkTask(config, client.mongoClient(), errorReporter);
+    RecordsAndExpectations recordsAndExpectations =
+        new RecordsAndExpectations(
+            asList(Records.simpleValid(TEST_TOPIC, 0), Records.simpleInvalid(TEST_TOPIC, 1)),
+            emptyList(),
+            emptyList());
+    assertThrows(RuntimeException.class, () -> task.put(recordsAndExpectations.records()));
+    recordsAndExpectations.assertExpectations(
+        client.capturedBulkWrites().get(DEFAULT_NAMESPACE),
+        errorReporter.reported(),
+        Records::match);
+  }
+
+  @Test
+  void putTolerateAllOrderedWriteError() {
+    properties.put(MongoSinkTopicConfig.ERRORS_TOLERANCE_CONFIG, ErrorTolerance.ALL.value());
+    MongoSinkConfig config = new MongoSinkConfig(properties);
+    client.configureCapturing(
+        DEFAULT_NAMESPACE,
+        collection ->
+            when(collection.bulkWrite(anyList(), any(BulkWriteOptions.class)))
+                // batch1
+                .thenReturn(BulkWriteResult.unacknowledged())
+                // batch2
+                .thenThrow(bulkWriteException(singletonList(2), false))
+                // batch2
+                .thenReturn(BulkWriteResult.unacknowledged()));
+    StartedMongoSinkTask task =
+        new StartedMongoSinkTask(config, client.mongoClient(), errorReporter);
+    RecordsAndExpectations recordsAndExpectations =
+        new RecordsAndExpectations(
+            asList(
+                // batch1
+                Records.simpleValid(TEST_TOPIC, 0),
+                // batch2
+                Records.simpleValid(TEST_TOPIC2, 1),
+                Records.simpleValid(TEST_TOPIC2, 2),
+                Records.simpleValid(TEST_TOPIC2, 3),
+                // batch3
+                Records.simpleValid(TEST_TOPIC, 4)),
+            asList(0, 1, 2, 3, 4),
+            emptyList());
```

Comment:
KAFKA-257 will change this empty expectation to `asList(2, 3)`:
- 2 will be reported because it has a corresponding write error;
- 3 will be reported because the bulk write is ordered, which means that 3 is not attempted to be written.

Reply:
Ack